src/main/java/de/photon/anticheataddition/user/data/TimestampMap.java

```diff
@@ -1,22 +1,18 @@
 package de.photon.anticheataddition.user.data;
 
-import java.util.EnumMap;
-import java.util.Map;
-
 public final class TimestampMap
 {
-    // As the map is not modified by a single thread, we can safely read without synchronization.
-    private final Map<TimeKey, Timestamp> map;
+    // We know that every enum has a non-null Timestamp as a value, so we can use an array instead of a map.
```

Comment:
FYI, an EnumMap also uses an array internally

Reply:
I know, but I would fill the EnumMap with the same Key for every value anyways, so that array is much more compact and easier to initialize.